src/Servers/Kestrel/Core/src/Internal/Http/MessageBody.cs

```diff
@@ -73,6 +73,17 @@ public virtual Task StopAsync()
 
         protected virtual Task OnStopAsync() => Task.CompletedTask;
 
+        public virtual void Reset()
```

Comment:
It would be good to call Reset from the constructor to ensure consistent state the first time it's used.